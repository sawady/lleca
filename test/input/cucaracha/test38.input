
fun f() : Int {
    return 65
}

fun main() {
    putChar(f())
    putChar(10)
}
