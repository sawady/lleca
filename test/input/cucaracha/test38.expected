Cons(
  Function(
    f,
    Nil,
    Int,
    Cons(
      StmtReturn(
        ExprConstNum(
          65
        )
      ),
      Nil
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprCall(
              f,
              Nil
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Nil
        )
      )
    ),
    Nil
  )
)
